server/src/testFixtures/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java

```diff
@@ -226,15 +231,16 @@ public static void waitForBlockOnAnyDataNode(String repository, TimeValue timeou
         assertThat(blocked).as("No repository is blocked waiting on a data node").isTrue();
     }
 
-    protected void createRepository(String repoName, String type) {
+    protected void createRepository(String repoName, String type) throws ExecutionException, InterruptedException {
         createRepository(repoName, type, randomRepositorySettings());
     }
 
-    protected void createRepository(String repoName, String type, Settings.Builder settings) {
+    protected void createRepository(String repoName, String type, Settings.Builder settings) throws ExecutionException, InterruptedException {
         logger.info("--> creating repository [{}] [{}]", repoName, type);
-        // CREATE REPOSITORY via SQL
-        assertAcked(client().admin().cluster().preparePutRepository(repoName)
-            .setType(type).setSettings(settings));
+        var putRepositoryRequest = new PutRepositoryRequest(repoName)
```

Comment:
didn't do it via SQL since "mock" repo type doesn't pass validation.
We have already test where we use mock type and we use request directly there.
https://github.com/crate/crate/commit/77fa9564274ba927c6c9db8562bf58303bbdd226#diff-4b4a49af01773be13ae3a5546b7ce700a427bcf1b42201156d8e7f00c3b99e41R66